src/makePageElement.ts

```diff
@@ -19,8 +19,11 @@ export async function getPageInfo({
   options: ExtendedOptions;
 }): Promise<PageInfo> {
   const pageObject = await getPageObject({ options });
-  let pageData = await fetchRouteData({ pageObject, options });
+  if (!pageObject.page) {
```

Comment:
Shall we add a sort of special property/flag to explicitly signal the pageObject type?

Reply:
Current approach takes advantage of Typescript's [discriminating-unions](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions) over the `page` property to correctly handle the cases when `page` is available and when it is not.

If we try to reuse the same 'PageObject` with the `page` defined we would probably need to use some `@ts-ignore` and "lie" to the compiler which I usually try to avoid.

What exactly did you have in mind here? Fell free to open a PR targeting this branch with your suggestion 🚀

Reply:
I didn't have in mind any TS trickery, TS is already tricky enough!

I wondered whether the way we currently tell `NotFoundPageObject` from `PageObject` by checking `page` prop could be more explicit.

Do you see any improvement if the discrimination between `NotFoundPageObject` and `PageObject` was something like a `type` prop instead of the presence of `page` prop?

Reply:
Do you think current check is not explicit enough?

I think that `if (!pageObject.page)` clearly indicates that there is no page found, but perhaps that just for me.

Were you thinking about something along these lines: `if(pageObject.type === 'notFound')`?

Reply:
Do you think current check is not explicit enough?

I think that `if (!pageObject.page)` clearly indicates that there is no page found, but perhaps that just for me.

Were you thinking about something along these lines: `if(pageObject.type === 'notFound')`?

Reply:
Do you think current check is not explicit enough?

I think that `if (!pageObject.page)` clearly indicates that there is no page found, but perhaps that's just me.

Were you thinking about something along these lines: `if(pageObject.type === 'notFound')`?

Reply:
Yep! It's not a game changer but I was wondering if setting an explicit type prop would make sense.

[Discriminating unions docs](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions) also use the same strategy.

Reply:
Added in https://github.com/toomuchdesign/next-page-tester/pull/135/commits/f22b2c06611af8f7eda206ea7569008807d3872a